packages/block-library/src/navigation/index.php

```diff
@@ -250,29 +250,24 @@ function block_core_navigation_render_submenu_icon() {
 
 
 /**
- * Finds the first non-empty `wp_navigation` Post.
+ * Finds the most recent published `wp_navigation` Post.
  *
  * @return WP_Post|null the first non-empty Navigation or null.
  */
-function block_core_navigation_get_first_non_empty_navigation() {
-	// Order and orderby args set to mirror those in `wp_get_nav_menus`
-	// see:
-	// - https://github.com/WordPress/wordpress-develop/blob/ba943e113d3b31b121f77a2d30aebe14b047c69d/src/wp-includes/nav-menu.php#L613-L619.
-	// - https://developer.wordpress.org/reference/classes/wp_query/#order-orderby-parameters.
+function block_core_navigation_get_most_recent_published_navigation() {
+	// We default to the most recently created
 	$parsed_args = array(
 		'post_type'      => 'wp_navigation',
 		'no_found_rows'  => true,
-		'order'          => 'ASC',
-		'orderby'        => 'name',
+		'order'          => 'DESC',
+		'orderby'        => 'date',
 		'post_status'    => 'publish',
-		'posts_per_page' => 20, // Try the first 20 posts.
+		'posts_per_page' => 1, // Try the first 20 posts.
 	);
 
-	$navigation_posts = new WP_Query( $parsed_args );
-	foreach ( $navigation_posts->posts as $navigation_post ) {
-		if ( has_blocks( $navigation_post ) ) {
-			return $navigation_post;
-		}
```

Comment:
Why did we ditch the `has_blocks` check here?

Reply:
If it's published and empty it is still valid :) IF the saving as draft idea works this makes even more sense.

Reply:
I just wonder _why_ we would have introduced this check if it served no purpose? 

When I saw it removed without explanation it triggered my spidey senses 😄

Reply:
I think it was that we _always_ wanted to show _something_ but with this new workflow it seems this is not the case. I hope removing it will not cause some weirdness, but since we want to respect editor and front parity of content, if the editor has a published empty menu and no items in it so will the front end, so this condition makes little sense.